### DISCUSSION TITLE
pg_net / net.http_post doesn’t support json arrays?

### DISCUSSION THREAD

**Discussion by liam-k** [supabase/supabase#17602] [Questions] [answered]
Hi! I’m currently using the http_post function of pg_net to post to an email API endpoint. (Specifically [this one](https://resend.com/docs/api-reference/emails/send-batch-emails), but it doesn’t really matter.) This endpoint wants a json array, not an object, which is generally possible in Javascript or something like CURL. However, pg_net seems to only support json objects. I’m getting the error 

_"Failed to run sql query: function net.http_post(unknown, jsonb[], headers => jsonb) does not exist"._ 

It works fine when having a ::jsonb type and object and not a ::jsonb[] array (but of course the API doesn’t like that). 

Am I overlooking something or should I submit this as a feature request?

This is the code just for illustration, really. Could be any kind of array or object and any kind of API endpoint. It doesn’t seem to be possible to supply an array.

```
SELECT
    net.http_post (
      'https://api.resend.com/emails/batch',

    [
      {
          "from": "Acme <onboarding@resend.dev>",
          "to": ["foo@gmail.com"],
          "subject": "hello world",
          "text": "it works!"
      },
      {
          "from": "Acme <onboarding@resend.dev>",
          "to": ["bar@gmail.com"],
          "subject": "hello world",
          "text": "it works!"
       }
    ]'::jsonb[],

      headers := '{"Authorization": "Bearer [api key]", "Content-Type": "application/json"}'::jsonb

    ) INTO request_id;
  RETURN request_id;
```

**Comment by steve-chavez** [comment, marked as answer]
Hi there,

Hm, why not use only `jsonb` instead? It will support the same value.

**Reply by liam-k** [reply]
Oh jeez. I’m quite sure I tried that but got some errors (probably because I was generating an array from a table and tried to use that). Anyway, I tried it with my minimal example and it does work, so that’s great (even though I already wrote a Javascript Edge Function in the meantime). Thank you!
